tests/foreman/api/test_rhcloud_inventory.py

```diff
@@ -78,24 +80,41 @@ def test_rhcloud_inventory_api_e2e(
         5. JSON files inside report can be parsed
         6. metadata.json lists all and only slice JSON files in tar
         7. Host counts in metadata matches host counts in slices
-        8. Assert Hostnames, IP addresses, and installed packages are present in report.
+        8. metadata contains source and foreman_rh_cloud_version keys.
+        9. Assert Hostnames, IP addresses, infrastructure_type, and installed packages
+            are present in report.
 
     :CaseImportance: Critical
 
-    :BZ: 1807829, 1926100, 1965234
+    :BZ: 1807829, 1926100, 1965234, 1824183, 1879453
     """
     org, ak = organization_ak_setup
     virtual_host, baremetal_host = rhcloud_registered_hosts
     local_report_path = robottelo_tmp_dir.joinpath(f'{gen_alphanumeric()}_{org.id}.tar.xz')
     # Generate report
-    generate_inventory_report(rhcloud_sat_host, org)
+    timestamp = datetime.utcnow().strftime('%Y-%m-%d %H:%M')
+    rhcloud_sat_host.api.Organization(id=org.id).rh_cloud_generate_report()
+    wait_for(
+        lambda: rhcloud_sat_host.api.ForemanTask()
+        .search(query={'search': f'{generate_report_task} and started_at >= "{timestamp}"'})[0]
+        .result
+        == 'success',
+        timeout=400,
+        delay=15,
+        silent_failure=True,
+        handle_exception=True,
+    )
     # Download report
     rhcloud_sat_host.api.Organization(id=org.id).rh_cloud_download_report(
         destination=local_report_path
     )
     common_assertion(local_report_path)
     # Assert Hostnames, IP addresses, and installed packages are present in report.
     json_data = get_report_data(local_report_path)
+    json_meta_data = get_report_metadata(local_report_path)
+
+    assert json_meta_data['source_metadata'] == {'foreman_rh_cloud_version': '4.0.27'}
```

Comment:
@adarshdubey-star Can you get the `tfm-rubygem-foreman_rh_cloud` package version from satellite and compare it with `json_meta_data['source_metadata']['foreman_rh_cloud_version']`?

Reply:
Also, I guess you forgot to add assertion for `infrastructure_type`.

Reply:
Noted!